lib/driver.js

```diff
@@ -232,14 +232,28 @@ class XCUITestDriver extends BaseDriver {
     await this.runReset();
 
     log.info(`Setting up ${this.isRealDevice() ? 'real device' : 'simulator'}`);
-    if (!this.isRealDevice()) {
+
+    if (this.isRealDevice()) {
+      if (this.opts.app) {
+        await this.installApp();
+      }
+    } else {
+      const installAppPromise = new Promise((resolve, reject) => {
```

Comment:
actually wondering if we should just resolve this immediately if we don't have an app, that way we don't even wait for the boot completed event?

```js
const installAppPromise = new Promise((resolve, reject) => {
  if (!this.opts.app) return resolve();
  this.opts.device.on(BOOT_COMPLETED_EVENT, async () => {
    try {
      await this.installApp();
      resolve();
    } catch (err) {
      reject(err);
    }
  });
});
```